odoo/models.py

```diff
@@ -4387,7 +4387,9 @@ def _parent_store_update(self):
         """
         cr.execute(query.format(self._table), [prefix, tuple(self.ids)])
         modified_ids = {row[0] for row in cr.fetchall()}
-        self.browse(modified_ids).modified(['parent_path'])
+        modified_records = self.browse(modified_ids)
+        modified_records.invalidate_cache(['parent_path'])
```

Comment:
Careful the `invalidate_cache` doesn't take `self` in account, you should give it `ids=modified_ids` (or you will invalidate too much). 

@rco-odoo  Also, maybe we should return the new parent_path from the update query (and put it in cache) to avoid an extra read after ?

Reply:
Should be even better to modify directly the cache to avoid a new query afterwards. :)

Reply:
Indeed, one option is to invalidate the cache on the given records with
```py
self.invalidate_cache(['parent_path'], ids=modified_ids)
```
The other option is to modify the `RETURNING` clause above to retrieve the new field value, and put that value in cache.